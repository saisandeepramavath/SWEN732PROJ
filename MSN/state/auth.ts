import {atom} from 'recoil'

import { FirebaseAuthTypes } from '@react-native-firebase/auth';

export const authState = atom<{
  isAuth: boolean;
  user: FirebaseAuthTypes.User | null;
}>({
  key: 'authState',
  default: {
    isAuth: false,
    user: null,
  },
});

export const userState = atom({
    key: 'userState',
    default: {
        name: '',
        email: '',
        photo: '',
        uid: '',
        groups: [],
    }
})

export const groupsState = atom({
    key: 'groupsState',
    default: [],
})
export const selectedGroupState = atom({
    key: 'selectedGroupState',
    default: null,
})
export const selectedGroupMembersState = atom({
    key: 'selectedGroupMembersState',
    default: [],
})
export const selectedGroupTransactionsState = atom({
    key: 'selectedGroupTransactionsState',
    default: [],
})