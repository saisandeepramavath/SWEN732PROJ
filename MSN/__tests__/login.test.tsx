import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import Index from '../app/index';
import auth from '@react-native-firebase/auth';

jest.mock('@react-native-firebase/auth');

describe('Login', () => {
	it('should match snapshot', () => {
		const { toJSON } = render(<Index />);
		expect(toJSON()).toMatchSnapshot();
	});

	it('should display error on failed login', async () => {
		const { getByPlaceholderText, getByText } = render(<Index />);
		fireEvent.changeText(getByPlaceholderText('Email'), 'john@example.com');
		fireEvent.changeText(getByPlaceholderText('Password'), 'wrongpassword');
		fireEvent.press(getByText('Login'));

		// await waitFor(() => {
		// 	expect(auth().signInWithEmailAndPassword).toHaveBeenCalledWith('john@example.com', 'wrongpassword');
		// 	expect(getByText('Sign in failed:')).toBeTruthy();
		// });
	});

	it('should navigate to home on successful login', async () => {
		(auth().signInWithEmailAndPassword as jest.Mock).mockResolvedValueOnce({});
		const { getByPlaceholderText, getByText } = render(<Index />);
		fireEvent.changeText(getByPlaceholderText('Email'), 'john@example.com');
		fireEvent.changeText(getByPlaceholderText('Password'), 'correctpassword');
		fireEvent.press(getByText('Login'));

		// await waitFor(() => {
		// 	expect(auth().signInWithEmailAndPassword).toHaveBeenCalledWith('john@example.com', 'correctpassword');
		// });
	});
});
