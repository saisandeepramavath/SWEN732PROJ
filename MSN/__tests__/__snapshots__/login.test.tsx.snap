// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "justifyContent": "center",
      "marginHorizontal": 20,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "paddingBottom": 0,
      }
    }
  >
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 32,
          "fontWeight": "bold",
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Welcome
    </Text>
    <TextInput
      autoCapitalize="none"
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#aaa"
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#ddd",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 50,
          "marginVertical": 10,
          "padding": 10,
        }
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#aaa"
      secureTextEntry={true}
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#ddd",
          "borderRadius": 8,
          "borderWidth": 1,
          "height": 50,
          "marginVertical": 10,
          "padding": 10,
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#007BFF",
          "borderRadius": 8,
          "marginVertical": 10,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#007BFF",
          "borderRadius": 8,
          "marginVertical": 10,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Create account
      </Text>
    </View>
  </View>
</View>
`;
